src/App.jsx

```diff
@@ -14,45 +15,81 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+export const App = () => {
+  const [currentOrder, setCurrentOrder] = useState([...goodsFromServer]);
+  const [isOriginalOrder, setIsOriginalOrder] = useState(true);
+
+  const sortAlphabetically = () => {
+    const newOrder = [...currentOrder].sort();
+
+    setCurrentOrder(newOrder);
+    setIsOriginalOrder(false);
+  };
+
+  const sortByLength = () => {
+    const newOrder = [...currentOrder].sort((a, b) => a.length - b.length);
+
+    setCurrentOrder(newOrder);
+    setIsOriginalOrder(false);
+  };
+
+  const reverseOrder = () => {
+    const newOrder = [...currentOrder].reverse();
+
+    setCurrentOrder(newOrder);
+    setIsOriginalOrder(!isOriginalOrder);
+  };
```

Comment:
It is better to perform all manipulations outside the App component and transfer here the already prepared array (sorted depending on the conditions) and output it.